src/nvim/regexp_nfa.c

```diff
@@ -4853,6 +4854,9 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   fast_breakcheck();
   if (got_int)
     return FALSE;
+  if (nfa_time_limit != NULL && profile_passed_limit(*nfa_time_limit)) {
+    return FALSE;
```

Comment:
@jamessan It's still ok to assign `true`/`false` to an INT, and historically that's what we've done.

 Although it could be argued it's misleading, I guess, but it's probably still a net win compared to using `int` even where only 2 values are allowed.